distributed/tests/test_worker.py

```diff
@@ -1576,6 +1576,34 @@ async def test_close_gracefully(c, s, a, b):
         assert_amm_transfer_story(key, b, a)
 
 
+@pytest.mark.slow
+@gen_cluster(client=True, Worker=Nanny)
+async def test_close_gracefully_no_suspicious_tasks(c, s, a, b):
+    workers = {a.worker_address: a, b.worker_address: b}
+    started = Event()
+    block_task = Event()
+
+    def blocking_task():
+        started.set()
+        block_task.wait()
+
+    fut = c.submit(blocking_task)
+    await started.wait()
+
+    to_close = s.tasks[fut.key].processing_on.address
+
+    async def close_gracefully():
+        asyncio.create_task(get_worker().close_gracefully())
+
+    c.submit(close_gracefully, workers=[to_close])
```

Comment:
this doesn't seem to reliably close_gracefully the worker

Reply:
I suggest to use `c.run` instead. Client.run is not registering the function as a proper "dask task" but is just executing it on the event loop.

if you use `Client.submit` you'll also get a `client.Future` back that releases the task again once it is deleted. It's very possible that you are running here into race condition where the task is scheduled and released before it got a chance to run. All of this nonsense is not performed when Client.run is used.

Reply:
yep that works not sure why I forgot about .run